docs/aws/README.md

```diff
@@ -1,46 +1,108 @@
-CLUSTER_NAME=<your_cluster>
-REGION=<your_region>
+CLUSTER_NAME=$USER-karpenter-demo
+AWS_DEFAULT_REGION=us-west-2
 ```
 
-### Create Karpenter IAM Resources
-This command will create IAM resources used by Karpenter. For production use, please review and restrict these permissions as necessary.
+### Create a Cluster
+Skip this step if you already have a cluster.
+```bash
+eksctl create cluster \
+--name ${CLUSTER_NAME} \
+--version 1.18 \
+--region ${AWS_DEFAULT_REGION} \
+--nodegroup-name karpenter-aws-demo \
+--node-type m5.large \
+--nodes 1 \
+--nodes-min 1 \
+--nodes-max 10 \
+--managed
+```
+
+### Create IAM Resources
+This command will create IAM resources used by Karpenter. We recommend using Cloud Formation and IAM Roles for Service Accounts to manage these permissions. For production use, please review and restrict these permissions for your use case.
 ```bash
-// TODO, point to github raw uri
 aws cloudformation deploy \
   --stack-name Karpenter \
   --template-file ./docs/aws/karpenter.cloudformation.yaml \
   --capabilities CAPABILITY_NAMED_IAM \
   --parameter-overrides OpenIDConnectIdentityProvider=$(aws eks describe-cluster --name ${CLUSTER_NAME} | jq -r ".cluster.identity.oidc.issuer" | cut -c9-)
 ```
 
-### Enable IRSA
+### Install Karpenter Controller and Dependencies
+Karpenter relies on cert-manager for certificates and prometheus for metrics.
+
+```bash
+./hack/quick-install.sh
+```
+
+### Enable IRSA and attach IAM Role to Service Account
 Enables IRSA for your cluster. This command is idempotent, but only needs to be executed once per cluster.
 ```bash
 eksctl utils associate-iam-oidc-provider \
---region ${REGION} \
+--region ${AWS_DEFAULT_REGION} \
 --cluster ${CLUSTER_NAME} \
 --approve
-```
 
-### Attach the Permissions
-```bash
 kubectl patch serviceaccount karpenter -n karpenter --patch "$(cat <<-EOM
 metadata:
   annotations:
-     eks.amazonaws.com/role-arn: arn:aws:iam::${AWS_ACCOUNT_ID}:role/KarpenterControllerRole
+    eks.amazonaws.com/role-arn: arn:aws:iam::${AWS_ACCOUNT_ID}:role/KarpenterControllerRole
 EOM
 )"
+
+kubectl delete pods -n karpenter -l control-plane=karpenter # Restart controller to load credentials
+```
+
+### Create a Provisoner
+Create a default Provisioner that launches nodes configured with cluster name, endpoint, and caBundle.
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: provisioning.karpenter.sh/v1alpha1
+kind: Provisioner
+metadata:
+  name: default
+spec:
+  cluster:
+    name: ${CLUSTER_NAME}
+    caBundle: $(aws eks describe-cluster --name ${CLUSTER_NAME} | jq ".cluster.certificateAuthority.data")
+    endpoint: $(aws eks describe-cluster --name ${CLUSTER_NAME} | jq ".cluster.endpoint")
+EOF
+kubectl get provisioner default -oyaml
 ```
-If you've already installed the Karpenter controller, you'll need to restart the pod to load the credentials.
+
+### Create some pods
+Create some dummy pods and observe logs
```

Comment:
Maybe add a warning here this will provision 5 instances which needs to be manually cleaned if not required?

Reply:
+1